src/compiler/scala/tools/reflect/ToolBoxFactory.scala

```diff
@@ -136,7 +136,11 @@ abstract class ToolBoxFactory[U <: JavaUniverse](val u: U) { factorySelf =>
           val currentTyper     = analyzer.newTyper(analyzer.rootContext(NoCompilationUnit, EmptyTree).make(expr2, owner))
           val withImplicitFlag = if (!withImplicitViewsDisabled) (currentTyper.context.withImplicitsEnabled[Tree] _) else (currentTyper.context.withImplicitsDisabled[Tree] _)
           val withMacroFlag    = if (!withMacrosDisabled) (currentTyper.context.withMacrosEnabled[Tree] _) else (currentTyper.context.withMacrosDisabled[Tree] _)
-          def withContext      (tree: => Tree) = withImplicitFlag(withMacroFlag(tree))
+          val withAccessMode   =
+            if (settings.YIgnoreAcessModifier.isSetByUser) (currentTyper.context.withAccessModeOff[Tree] _)
```

Comment:
It should be simply `if (settings.YIgnoreAcessModifier)`, I think. `isSetByUser` doesn't indicate that the flag is set to `true`, but only that it has been set by user (even if the value is `false`).